### ISSUE TITLE
[10.0] auth_session_timeout causing server error with auth_cas module

### ISSUE THREAD

**Issue by remi-filament** [OCA/server-tools#1532]
Following merge of #1495 in server-tools (commit 5fa2cc9c9d05fa5be4aa7e21ac434b52bf6b2843), using both [auth_cas module from Elico_Corp](https://github.com/Elico-Corp/odoo-addons/tree/10.0/auth_cas) cause internal server error.

I did not manage to find out how to fix that issue, but managed to figure out at which time the issue appeared (on Monday 25th of March) and this is the only module that I am using which was updated on that day.

Server running on docker with latest OCB 10.0 code and docker image generated daily from [here](https://github.com/lefilament/odoo_docker)

Error log is reported below:

```
testodoo | 2019-03-28 14:55:48,710 1 INFO filament10_test werkzeug: 172.29.0.2 - - [28/Mar/2019 14:55:48] "GET /web HTTP/1.1" 303 -
testodoo | 2019-03-28 14:55:48,823 1 INFO filament10_test werkzeug: 172.29.0.2 - - [28/Mar/2019 14:55:48] "GET /web/login HTTP/1.1" 200 -
testodoo | 2019-03-28 14:55:49,070 1 INFO filament10_test werkzeug: 172.29.0.2 - - [28/Mar/2019 14:55:49] "GET /web/webclient/qweb?mods= HTTP/1.1" 304 -
testodoo | 2019-03-28 14:55:49,071 1 INFO filament10_test werkzeug: 172.29.0.2 - - [28/Mar/2019 14:55:49] "POST /web/webclient/bootstrap_translations HTTP/1.1" 200 -
testodoo | 2019-03-28 14:55:49,110 1 INFO filament10_test werkzeug: 172.29.0.2 - - [28/Mar/2019 14:55:49] "POST /web/webclient/translations HTTP/1.1" 200 -
testodoo | 2019-03-28 14:55:50,245 1 INFO filament10_test werkzeug: 172.29.0.2 - - [28/Mar/2019 14:55:50] "GET /auth_cas/cas_authenticate HTTP/1.1" 302 -
testodoo | 2019-03-28 14:55:58,619 1 INFO filament10_test werkzeug: 172.29.0.2 - - [28/Mar/2019 14:55:58] "GET /?ticket=ST-afa261e15a52dd3b46e60799a6c03e4146895b0189aff6eda26665fb7a3710cb HTTP/1.1" 500 -
testodoo | 2019-03-28 14:55:58,683 1 ERROR filament10_test werkzeug: Error on request:
testodoo | Traceback (most recent call last):
testodoo |   File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 193, in run_wsgi
testodoo |     execute(self.server.app)
testodoo |   File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 181, in execute
testodoo |     application_iter = app(environ, start_response)
testodoo |   File "/opt/odoo/odoo/odoo/service/server.py", line 307, in app
testodoo |     return self.app(e, s)
testodoo |   File "/opt/odoo/odoo/odoo/service/wsgi_server.py", line 184, in application
testodoo |     return werkzeug.contrib.fixers.ProxyFix(application_unproxied)(environ, start_response)
testodoo |   File "/usr/local/lib/python2.7/dist-packages/werkzeug/contrib/fixers.py", line 152, in __call__
testodoo |     return self.app(environ, start_response)
testodoo |   File "/opt/odoo/additional_addons/auth_brute_force/models/res_users.py", line 27, in _patch
testodoo |     return original_fn(environ, start_response)
testodoo |   File "/opt/odoo/additional_addons/auth_brute_force/models/res_users.py", line 27, in _patch
testodoo |     return original_fn(environ, start_response)
testodoo |   File "/opt/odoo/odoo/odoo/service/wsgi_server.py", line 172, in application_unproxied
testodoo |     result = handler(environ, start_response)
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 1326, in __call__
testodoo |     return self.dispatch(environ, start_response)
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 1300, in __call__
testodoo |     return self.app(environ, start_wrapped)
testodoo |   File "/usr/local/lib/python2.7/dist-packages/werkzeug/wsgi.py", line 599, in __call__
testodoo |     return self.app(environ, start_response)
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 1499, in dispatch
testodoo |     result = ir_http._dispatch()
testodoo |   File "/opt/odoo/odoo/addons/web_editor/models/ir_http.py", line 21, in _dispatch
testodoo |     return super(IrHttp, cls)._dispatch()
testodoo |   File "/opt/odoo/odoo/addons/utm/models/ir_http.py", line 20, in _dispatch
testodoo |     response = super(IrHttp, cls)._dispatch()
testodoo |   File "/opt/odoo/odoo/odoo/addons/base/ir/ir_http.py", line 199, in _dispatch
testodoo |     return cls._handle_exception(e)
testodoo |   File "/opt/odoo/odoo/odoo/addons/base/ir/ir_http.py", line 169, in _handle_exception
testodoo |     return request._handle_exception(exception)
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 775, in _handle_exception
testodoo |     return super(HttpRequest, self)._handle_exception(exception)
testodoo |   File "/opt/odoo/odoo/odoo/addons/base/ir/ir_http.py", line 195, in _dispatch
testodoo |     result = request.dispatch()
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 834, in dispatch
testodoo |     r = self._call_function(**self.params)
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 334, in _call_function
testodoo |     return checked_call(self.db, *args, **kwargs)
testodoo |   File "/opt/odoo/odoo/odoo/service/model.py", line 101, in wrapper
testodoo |     return f(dbname, *args, **kwargs)
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 327, in checked_call
testodoo |     result = self.endpoint(*a, **kw)
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 942, in __call__
testodoo |     return self.method(*args, **kw)
testodoo |   File "/opt/odoo/odoo/odoo/http.py", line 507, in response_wrap
testodoo |     response = f(*args, **kw)
testodoo |   File "/opt/odoo/additional_addons/auth_cas/controllers/main.py", line 135, in index
testodoo |     config.get('port'), config.get('auto_create'), ticket)
testodoo |   File "/opt/odoo/additional_addons/auth_cas/controllers/main.py", line 104, in cas_authenticate_user
testodoo |     user_id.write({'cas_key': cas_key, 'password': cas_key})
testodoo |   File "/opt/odoo/odoo/addons/hr/models/res_users.py", line 17, in write
testodoo |     result = super(User, self).write(vals)
testodoo |   File "/opt/odoo/odoo/addons/mail/models/res_users.py", line 57, in write
testodoo |     write_res = super(Users, self).write(vals)
testodoo |   File "/opt/odoo/odoo/odoo/addons/base/res/res_users.py", line 864, in write
testodoo |     res = super(UsersView, self).write(values)
testodoo |   File "/opt/odoo/odoo/odoo/addons/base/res/res_users.py", line 697, in write
testodoo |     res = super(UsersImplied, self).write(values)
testodoo |   File "/opt/odoo/odoo/odoo/addons/base/res/res_users.py", line 382, in write
testodoo |     self._invalidate_session_cache()
testodoo |   File "/opt/odoo/odoo/odoo/addons/base/res/res_users.py", line 547, in _invalidate_session_cache
testodoo |     self._compute_session_token.clear_cache(self)
testodoo | AttributeError: 'function' object has no attribute 'clear_cache'
```

**Comment by pedrobaeza** [comment]
Sorry, but the other module is not an OCA one, so we can't provide support on it.

**Comment by remi-filament** [comment]
Thank you @pedrobaeza, I know the other module is not an OCA one, and I am not asking support for it.
The setup worked for months until a regression was introduced by updating auth_session_timeout beginning of this week.
Disabling  auth_session_timeout solves the issue, so I really believe that a regression was introduced which should be handled, which is why I report it here.

**Comment by pedrobaeza** [comment]
No, it's not a regression, but an incompatibility of both modules. `auth_session_timeout` works as expected with all my known OCA modules, but not with that one, so the possibilities are:

1. The other module is not well developed and introduces a problem that is not shown until the last change here (but existed before).
2. The change in OCA module has a compatibility problem and then, this case resides in OCA side, but I doubt it due to the QA we have in OCA + the change has been done and reviewed by well-known reputed contributors, although nobody is exempted from a mistake, of course. Please tell me if it's this one and I'll reopen the issue.
3. It's nobody fault, but just an incompatibility between both due to the part they are touching. A solution should be found for compatibilizing both.

**Comment by remi-filament** [comment]
@pedrobaeza
My purpose is clearly not to blame anyone, just to report on a possible issue.
I tested on another environment without auth_cas module present and I managed to reach the same error by modifying password from User Form view as an administrator (in configuration > Users).
Would you mind checking if you can reproduce that way ?

**Comment by pedrobaeza** [comment]
OK, then it would be point 2. Reopening.

@daramousk @hbrunn can you please check what's happening?

**Comment by hbrunn** [comment]
@daramousk you need to reproduce the caching decorator of the original in https://github.com/OCA/server-tools/pull/1495/files#diff-67171e8ae8a4e346f4c6f4015b4a2b68R105
(cf https://github.com/OCA/OCB/blob/10.0/odoo/addons/base/res/res_users.py#L525) - decorators aren't inherited

**Comment by pedrobaeza** [comment]
I have fixed it directly in 7b33aca2fa87cd853c9b5ec4edca2adf42447994

**Comment by remi-filament** [comment]
Thanks, it works !
